### PR TITLE
Make HazelcastJsonValue work with PagingPredicate

### PR DESCRIPTION
**Pull request by sancar**
HazelcastJsonValue is made Comparable to work with PagingPredicate

fixes https://github.com/hazelcast/hazelcast/issues/14888

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/core/HazelcastJsonValue.java [hunk 1]
```diff
@@ -35,7 +37,7 @@
  * Ill-formatted JSON strings may cause false positive or false negative
  * results in queries. {@code null} string is not allowed.
  */
-public final class HazelcastJsonValue {
+public final class HazelcastJsonValue implements Comparable<HazelcastJsonValue> {
```
Comment (srknzl) [inline hazelcast/src/main/java/com/hazelcast/core/HazelcastJsonValue.java:40]: With this change all languages that has HazelcastJsonValue will be able use PagingPredicate with HazelcastJsonValue, is it right?

**Review by srknzl — APPROVED** [review]
I think it looks good. 

I believe we should document the sorting behaviour in the API documentation and reference manual, possibly [here](https://docs.hazelcast.com/hazelcast/5.0/query/querying-maps-predicates#querying-json-strings) in the reference manual.

File: hazelcast/src/main/java/com/hazelcast/core/HazelcastJsonValue.java (hunk 1 above)
Comment (sancar) [reply hazelcast/src/main/java/com/hazelcast/core/HazelcastJsonValue.java:40]: Yes, that is right.

**Comment by mdumandag** [comment]
I think it might be good to have some ideas regarding this from the SQL team as well. @ivanthescientist can you comment on this?

**Comment by ivanthescientist** [comment]
I don't think this will affect SQL in any way. Every operation related to JSON will extract String value first anyway.

**Review by ihsandemir — APPROVED** [review]

**Pull request merged**